plugins/router/freebox

```diff
@@ -0,0 +1,316 @@
+#!/bin/bash
+# -*- sh -*-
+
+: << =cut
+
+=head1 NAME
+
+freebox - Plugin to monitor stats of a Freebox
+
+The ideal usage is also used as an updated warning limit.
+
+=head1 MAGIC MARKERS
+
+  #%# family=auto
+  #%# capabilities=autoconf
+
+=head1 CAVEAT
+
+Only tested on a Freebox v5 with an ADSL uplink.
+
+=head1 AUTHOR
+
+Olivier Mehani
+
+Copyright (C) 2019 Olivier Mehani <shtrom+munin@ssji.net>
+
+=head1 LICENSE
+
+SPDX-License-Identifier: GPL-3.0-or-later
+
+=cut
+
+# shellcheck disable=SC1090
+. "${MUNIN_LIBDIR}/plugins/plugin.sh"
+
+if [ "${MUNIN_DEBUG}" = 1 ]; then
+    set -x
+fi
+
+for CMD in curl iconv bc; do
+	if ! command -v "${CMD}" >/dev/null; then
+		echo "${CMD} not found" >&2
+		exit 1
+	fi
+done
+
+fbx_info_url="http://mafreebox.free.fr/pub/fbx_info.txt"
+
+fetch() {
+	# shellcheck disable=SC2154
+	curl -s "$@"
+}
+
+# XXX: testing
+# INFO="$( iconv -f latin1 -t utf8 fbx_info.txt)"
+INFO="$(fetch "${fbx_info_url}" | iconv -f latin1 -t utf8)"
+
+eval $(
+	sed -nE '
+		s#^\s*Temps depuis la mise en route\s*(([0-9]+) jours?, )?(([0-9]+) heures?, )?(([0-9]+) minutes?)#UPTIME_DAYS="0\2" UPTIME_HOURS="0\4" UPTIME_MINUTES="0\6"#p;
+		s#^\s*Débit ATM\s*([0-9]+)\s*kb/s\s*([0-9]+)\s*kb/s#ATM_DOWN="0\1" ATM_UP="0\2"#p;
+		s#^\s*Marge de bruit\s*([0-9.]+)\s*dB\s*([0-9.]+)\s*dB#NOISE_DOWN="0\1" NOISE_UP="0\2"#p;
+		s#^\s*Atténuation\s*([0-9.]+)\s*dB\s*([0-9.]+)\s*dB#ATTENUATION_DOWN="0\1" ATTENUATION_UP="0\2"#p;
+		s#^\s*FEC\s*([0-9]+)\s*\s*([0-9]+)\s*#FEC_DOWN="0\1" FEC_UP="0\2"#p;
+		s#^\s*CRC\s*([0-9]+)\s*\s*([0-9]+)\s*#CRC_DOWN="0\1" CRC_UP="0\2"#p;
+		s#^\s*HEC\s*([0-9]+)\s*\s*([0-9]+)\s*#HEC_DOWN="0\1" HEC_UP="0\2"#p;
+		s#^\s*WAN.*\s([0-9]+)\s*ko/s\s*([0-9]+)\s*ko/s#WAN_DOWN="0\1" WAN_UP="0\2"#p;
+		s#^\s*Ethernet.*\s([0-9]+)\s*ko/s\s*([0-9]+)\s*ko/s#ETH_DOWN="0\1" ETH_UP="0\2"#p;
+		s#^\s*USB.*\s([0-9]+)\s*ko/s\s*([0-9]+)\s*ko/s#USB_DOWN="0\1" USB_UP="0\2"#p;
+		s#^\s*Switch.*\s([0-9]+)\s*ko/s\s*([0-9]+)\s*ko/s#SWITCH_DOWN="0\1" SWITCH_UP="0\2"#p;
+	' <<< "${INFO}"
+)
+
+DHCP_CLIENTS=$(sed -nE '/Attributions dhcp/,/Redirections de ports/{s/^\s*([0-9A-F:]{17}).*$/\1/p}' <<< "${INFO}" | wc -l)
+REDIRECT_TCP=$(sed -nE '/Redirections de ports/,/Interfaces réseau/{s/^\s*(TCP).*$/\1/p}' <<< "${INFO}" | wc -l)
+REDIRECT_UDP=$(sed -nE '/Redirections de ports/,/Interfaces réseau/{s/^\s*(UDP).*$/\1/p}' <<< "${INFO}" | wc -l)
+
+FREEBOX_UPTIME=$(echo "${UPTIME_DAYS}*86400+${UPTIME_HOURS}*3600+${UPTIME_MINUTES}*60" | bc -ql)
+LAST_CONNECT=$(sed -nE '/Journal de connexion adsl/,+5{s#^\s*([0-9]{2})/([0-9]{2})/([0-9]{4})[^0-9]*([0-9]+):([0-9]+):([0-9]+)\s+Connexion.*#\3-\2-\1T\4:\5:\6#p}' <<< "${INFO}")
+CONNECTION_UPTIME=na
+if [ -n "${LAST_CONNECT}" ]; then
+	FORMULA="$(date +%s)-$(date -d "${LAST_CONNECT}" +%s)"
```

Comment:
I could imagine, that the following is as readable as your suggestion (without requiring `bc`):
```
FORMULA=$(( $(date +%s) - $(date -d "$LAST_CONNECT" + %s) ))